### DISCUSSION TITLE
Sensor that uses resources

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8556] [Q&A]
*Sensor that uses resources*

I'd like to create a sensor that triggers a job for each email in an inbox matching certain criteria.
I have already created a resource which my ops use to interact with a gmail account. That gmail resource is itself dependent on another resource which supplies authentication credentials. I'd like to reuse these resources with my sensor, but it seems like `@sensor` doesn't support providing resources.

Any tips on reusing resources with sensors?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1646630730596519?thread_ts=1646630730.596519&cid=C01U954MEER).*

**Comment by yuhan** [comment]
You can use the [build_resources](https://docs.dagster.io/_apidocs/resources#dagster.build_resources) API to initialize a resource within your sensor.

See details and example in the doc: https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#using-resources-in-sensors
